CondCore/JetMETPlugins/plugins/JetResolution_PayloadInspector.cc

```diff
@@ -293,48 +383,117 @@ namespace JME {
+      std::string stmp;
+
+      std::string tag_ver;
+      std::string tag_res;
+      std::string tag_jet;
+
+      getline(ss_tagname, stmp, '_');  // drop first
+      getline(ss_tagname, stmp, '_');  // year
+      tag_ver = stmp;
+      getline(ss_tagname, stmp, '_');  // ver
+      tag_ver += '_' + stmp;
+      getline(ss_tagname, stmp, '_');  // cmssw
+      tag_ver += '_' + stmp;
+      getline(ss_tagname, stmp, '_');  // data/mc
+      tag_ver += '_' + stmp;
+      getline(ss_tagname, stmp, '_');  // bin
+      tag_res = stmp;
+      getline(ss_tagname, stmp, '_');  // jet algorithm
+      tag_jet = stmp;
+
+      bool is_2bin = false;
 
       if (payload.get()) {
         if (!payload->getRecords().empty() &&  // No formula for SF
             payload->getDefinition().getFormulaString().compare("") != 0)
           return false;
 
+        is_2bin = false;
         for (const auto& record : payload->getRecords()) {
+          if (record.getBinsRange().size() > 1)
+            is_2bin = true;
+
           if (!record.getBinsRange().empty() && payload->getDefinition().getBinName(0) == "JetEta" &&
               record.getParametersValues().size() == 3) {  // norm, down, up
 
+            //               payload->getDefinition().getBinName(1) == "JetPt" &&
+            //               !record.getBinsRange()[1].is_inside(500.) )
+            //              continue; // for 2-bin payload, take jetpt=500
+
             for (size_t it = 0; it <= NBIN_ETA; it++) {
               double x_axis = (it + 0.5) * (MAX_ETA - MIN_ETA) / NBIN_ETA + MIN_ETA;
-              if (record.getBinsRange()[0].is_inside(x_axis)) {
+              if (((is_2bin == false) || (is_2bin == true && record.getBinsRange()[1].is_inside(500.))) &&
+                  record.getBinsRange()[0].is_inside(x_axis)) {
+                //edm::LogWarning("JRObj_PI") << "JetEta " << x_axis << ", SF "
+                // << record.getParametersValues()[0];
                 sf_eta_norm->SetBinContent(it + 1, record.getParametersValues()[0]);
                 sf_eta_down->SetBinContent(it + 1, record.getParametersValues()[1]);
                 sf_eta_up->SetBinContent(it + 1, record.getParametersValues()[2]);
               }
             }
           }
-        }  // records
+
+          if (record.getBinsRange().size() > 1 && payload->getDefinition().getBinName(0) == "JetEta" &&
+              record.getBinsRange()[0].is_inside(2.55) &&  // take jeteta=2.5
+              payload->getDefinition().getBinName(1) == "JetPt" &&
+              record.getParametersValues().size() == 3) {  // norm, down, up
+
+            is_2bin = true;
+
+            for (size_t it = 0; it <= NBIN_PT; it++) {
+              double x_axis = (it + 0.5) * (MAX_PT - MIN_PT) / NBIN_PT + MIN_PT;
+              if (record.getBinsRange()[1].is_inside(x_axis)) {
+                //edm::LogWarning("JRObj_PI") << "JetPt " << x_axis << ", SF " << record.getParametersValues()[0];
```

Comment:
please remove commented out code, here actually you could make it LogDebug right?

Reply:
Yes.